docs/src/main/asciidoc/_developer-guide/ejb3/EJB_on_Kubernetes.adoc

```diff
@@ -0,0 +1,104 @@
+The persistent storage over restarts is needed as the transaction log is usually stored at the file system.
+If the transaction manager creates a transaction log record it's created only at the transaction log particular to the WildFly instance.
+The hostname stability is needed as the WildFly may be contacted via EJB remote call with transaction propagation.
+The WildFly has to be reachable under the same hostname even after pod restarts.
+As the transaction log is bound to the particular WildFly instance it may be finished only there.
+
+
+[[ejb-calls-on-kubernetes]]
+== EJB calls on Kubernetes
+
+The EJB caller has two options on how to configure the remote call.
+It can be defined either as a remote outbound connection (see details at link:Admin_guide{outfilesuffix}#outbound-connections[Admin Guide, section Outbound Connections])
+or you may use a direct `InitialContext` lookup in your code.
+
+If you use either case then for the Kubernetes you need to adjust the configuration of the target node.
+For the target hostname, you need to use the DNS name of the very first pod managed by `StatefulSet`.
+
+The `StatefulSet` guarantees depend on the ordering of the pods. The pods are named in the prescribed order.
+If you scale your application up to 3 replicas you may expect to name of pods like `wildfly-server-0`, `wildfly-server-1`, `wildfly-server-2`.
+
+It's expected a link:https://kubernetes.io/docs/concepts/services-networking/service/#headless-services[headless services]
+to be used along with the `StatefulSet`. With the headless service, there is ensured the DNS hostname for the pod.
+If the WildFly is run on `StatefulSet` with the headless service named `wildfly-server-headless`.
+Then the DNS name of the very first pod will be `wildfly-server-0.wildfly-server-headless`.
+
+If the Wildfly is run as cluster then this use of the hostname `wildfly-server-0.wildfly-server-headless`
+guarantees that the EJB call may reach any WildFly instance connected to the cluster.
+It's a bootstrap connection which is used to initialize the EJB client
+which gathers the structure of the WildFly cluster as the next step.
+
+
+[[ejb-kubernetes-configuration]]
+== EJB configuration for Kubernetes
+
+These are steps you need to process in order to run EJB remote calls.
+Some steps are related to the server configuration, the other ones
+on your application.
+
+. Configure the WildFly cluster on Kubernetes.
+  See the link:High_Availability_Guide{outfilesuffix}#discovery-for-kubernetes[High Availability Guide, section of Kubernetes discovery].
+. All the `socket-binding` must have defined a client mapping for the DNS value mapped by `StatefulSet` headless service.
+  For example in the application which is named `wildfly-server` and considering the `StatefulSet` headless service is named `wildfly-server-headless`
+  then the `http` socket binding needs to be defined in the following way
+
+  ```
```

Comment:
This is markdown, not asciidoc notation.

https://asciidoctor.org/docs/asciidoc-writers-guide/#listing-and-source-code-blocks